tracr/actions/pcr.py

```diff
@@ -0,0 +1,103 @@
+from scipy import stats
+from matplotlib import pyplot as plt
+from mpl_toolkits.mplot3d import Axes3D
+import sys, os, glob
+import numpy as np
+import json
+
+def scrape_data(all_build_data=None, plate_list=['2'],
+                 feat_list=None, resp_var=None, stat_param=None):
+
+    # Retrieve all build features
+    if all_build_data==None:
+        try:
+            all_build_data = np.genfromtxt('process_parameters.csv', delimiter=',',
+                                             dtype=None)
+        except IOError:
+            print 'Please input a process_parameters file'
+
+    # Obtain build features of interest
+    if feat_list==None:
+        print 'Please input a feature list with keys matching p_p file'
+    idx = {}
+    idx['plate'] = np.argwhere(all_build_data[0,:]=='plate')[0][0]
+    for feat in feat_list:
+        idx[feat] = np.argwhere(all_build_data[0,:]==feat)[0][0]
+
+     # Obtain user specified defect parameter for regression
+    if resp_var==None:
+        print json.loads(open('A13_pore-distribution.json', 'rb').read()).keys()
+        resp_var = raw_input('Please enter (void specific) response variable: ')
+    if stat_param==None:
+        stat_param = raw_input("Please enter statistical metric (mean, median, mode): ")
+
+    # Iterate through .json files for response parameters
+    processed_parts = []
+    response = []
+    for f in glob.glob('*.json'):
+        data = json.loads(open(f, 'rb').read())
+        ifile = os.path.basename(f)
+        processed_parts.append(os.path.splitext(ifile)[0][:3])
+        if stat_param=='mean':
+            response.append(np.mean(data[resp_var]['values']))
+        elif stat_param=='mode':
+            response.append(stats.mode(part[resp_var]['values'],axis=None)[0][0])
+        else:
+            response.append(np.median(data[resp_var]['values']))
+    np.reshape(processed_parts, (len(processed_parts),1))
+    np.reshape(response, (len(response),1))
+
+    # Remove zeros from part names in 'processed_parts'
+    abbrev_parts = []
+    for j in processed_parts:
+        if j[1]=='0':
+            abbrev_parts.append(j[0]+j[2])
+        else:
+            abbrev_parts.append(j)
+    np.reshape(abbrev_parts,(len(abbrev_parts),1))
+
+    # Generate ID-response array
+    input_data = []
+    print np.asarray(abbrev_parts)
+    print np.asarray(response)
+    ID_response = np.vstack((np.asarray(abbrev_parts).T, np.asarray(response).T)).T
+    for sample in all_build_data:
+        if any(y==sample[idx['plate']] for y in plate_list):
+            if (any(x==(sample[idx['col']]+sample[idx['row']]) for x in abbrev_parts)):
+                match = np.argwhere((sample[idx['col']]+sample[idx['row']])==
+                                        ID_response[:,0])
+                sample = np.append(sample, ID_response[match,1:])
+                input_data.append(sample)
+        input_data = np.vstack(input_data)
+    return ID_response
+
+def pcr(data):
```

Comment:
Was using that return for testing earlier, has been replaced with input_data


--
Henry




> On Aug 29, 2016, at 3:02 PM, bkappes <notifications@github.com> wrote:
> 
> In tracr/actions/pcr.py <https://github.com/csm-adapt/tracr/pull/8#discussion_r76686795>:
> 
> > +    # Generate ID-response array
> > +    input_data = []
> > +    print np.asarray(abbrev_parts)
> > +    print np.asarray(response)
> > +    ID_response = np.vstack((np.asarray(abbrev_parts).T, np.asarray(response).T)).T
> > +    for sample in all_build_data:
> > +        if any(y==sample[idx['plate']] for y in plate_list):
> > +            if (any(x==(sample[idx['col']]+sample[idx['row']]) for x in abbrev_parts)):
> > +                match = np.argwhere((sample[idx['col']]+sample[idx['row']])==
> > +                                        ID_response[:,0])
> > +                sample = np.append(sample, ID_response[match,1:])
> > +                input_data.append(sample)
> > +        input_data = np.vstack(input_data)
> > +    return ID_response
> > +
> > +def pcr(data):
> @sinistral <https://github.com/sinistral> , @HGeerlings <https://github.com/HGeerlings> 
> What do you think about the return type here? Perhaps this best returns a functor that takes a vector of feature values and returns a predicted response?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/csm-adapt/tracr/pull/8/files/ce1f0326a830d5eda0d2c40b4925ff997b1970c9..0e789d0b796025a56d8f9ff7715edf7236627273#r76686795>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AMZvXTxnjSqrgiYCkqoQ-8a5zwaWwCoVks5qk0j7gaJpZM4JvDIc>.
>